packages/react/src/modules/slack/components/SlackChannelCombobox/SlackChannelCombobox.tsx

```diff
@@ -72,21 +79,24 @@ export const SlackChannelCombobox = ({
     updating: connectedChannelsUpdating,
   } = useConnectedSlackChannels({ slackChannelsRecipientObject });
 
-  const [comboboxListOpen, setComboboxListOpen] = useState<boolean>(false);
-  const [inputValue, setInputValue] = useState("");
-
-  // Used to close the combobox when clicking outside of it
-  const comboboxRef = useRef(null);
-  useOutsideClick({ ref: comboboxRef, fn: () => setComboboxListOpen(false) });
-
-  // Used to optimistically show when user toggles a channel connected or not
   const [currentConnectedChannels, setCurrentConnectedChannels] = useState<
     SlackChannelConnection[] | null
   >(null);
 
   useEffect(() => {
-    setCurrentConnectedChannels(connectedChannels);
-  }, [connectedChannels]);
+    // Used to make sure we're only showing currently available channels to select from.
+    // There are cases where a channel is "connected" in Knock, but it wouldn't be
+    // posting to it if the channel is private and the Slackbot doesn't belong to it,
+    // so the channel won't show up here and it won't be posted to.
+    const channels =
+      connectedChannels?.filter((connectedChannel) => {
+        return !!slackChannels.find(
+          (slackChannel) => slackChannel.id === connectedChannel.channel_id,
```

Comment:
We might want to consider grouping the slack channels here by id so this lookup can be done in constant time rather than traversing the list